src/project-context.ts

```diff
@@ -14,12 +15,19 @@ export class ProjectContext {
 
   static getAccountId(scope: Construct): string {
     const account = ProjectContext.getProjectAccountConfiguration(scope);
+    if (!account.id) {
+      throw new Error(`Account ${this.getName(scope)} does not have an ID`);
+    }
     return account.id;
   }
 
-  static getAccountConfig(scope: Construct, key: string): any {
+  static getAccountConfig(scope: Construct, key: string, defaultValue?: any): any {
     const account = ProjectContext.getProjectAccountConfiguration(scope);
-    return account.config?.[key];
+    const value = get(account.config, key, defaultValue);
+    if (value === undefined || value === '') {
+      throw new Error(`Account ${this.getName(scope)} does not have a config with key ${key}`);
+    }
+    return value;
```

Comment:
Instead of throwing an error, the "more CDK way" seems to be setting an error annotation into the scope. 

This works by importing `Annotations` from `aws-cdk-lib` and then using it to add "Error" annotation for given scope: `Annotations.of(scope).addError(message);`. The not-so-great thing is that from code perspective not throwing means one has to return some empty/undefined value, but that is okay as CDK will stop synthetization once it encounters an Error annotation. 

You can use: https://github.com/almamedia-open-source/cdk-project-context/blob/main/src/error.ts See [example usage](https://github.com/almamedia-open-source/cdk-project-context/blob/1b466e55771c54b1dd836a6284a8a04f5b8c051d/src/environment-type.ts#L30-L32).

Reply:
Reason to use Annotation is that end-user will receive an error message pointing to the given scope (path), so they'll know exactly where the error was.